be/src/exprs/like_predicate.cpp

```diff
@@ -172,6 +193,20 @@ Status LikePredicate::regex_prepare(FunctionContext* context, FunctionContext::F
     if (!context->is_notnull_constant_column(1)) {
         return Status::OK();
     }
+    if (context->get_num_args() == 3 && context->is_notnull_constant_column(2)) {
+        ColumnPtr column_ignore_case = context->get_constant_column(2);
```

Comment:
If the third arg is not a constant, then this branch is skipped. If you want ensure the third arg is special value, you'd better add these check in analyze phase in FE. please reference `FunctionAnalyzer.java`

Reply:
If FE has checked, this can be deleted.